### PR TITLE
REF: remove _convert_scalar_indexer

### PR DESCRIPTION
**Pull request by jbrockmendel**
This sits on top of #31867, so is partially a demonstration of how much complication is caused by our inconsistent error-raising.

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
@jbrockmendel can you merge master

**Comment by simonjayhawkins** [comment]
xref #25996

**Comment by WillAyd** [comment]
Definitely simplifies things so +1 there. Does this need to get bundled with the other PR to comprehensively raise a `KeyError`? Again on board, just want to be careful of messaging and backwards compat for users that previously caught TypeError (if even an issue - maybe they always caught both?)

**Comment by jbrockmendel** [comment]
> Does this need to get bundled with the other PR to comprehensively raise a KeyError?

Yes.

**Comment by jbrockmendel** [comment]
closing to clear the queue, will re-open following #31867.

**Comment by jbrockmendel** [comment]
re-opened following #31867

File: pandas/core/series.py [hunk 1]
```diff
@@ -849,9 +849,7 @@ def __getitem__(self, key):
             return self
 
         key_is_scalar = is_scalar(key)
-        if key_is_scalar:
-            key = self.index._convert_scalar_indexer(key, kind="getitem")
-        elif isinstance(key, (list, tuple)):
+        if not key_is_scalar and isinstance(key, (list, tuple)):
```
Comment (simonjayhawkins) [inline pandas/core/series.py]: do we need the `not key_is_scalar` and is there some overlap with the check in L872-875

Comment (jbrockmendel) [reply pandas/core/series.py]: will remove the redundant part of the check; will hold off on more invasive refactor

Comment (simonjayhawkins) [reply pandas/core/series.py]: probably no need then if refactoring later. I guess this is more performant at the expense of clarity. This method looks like it just needs one if key_is_scalar and an else.

**Comment by jreback** [comment]
love it. if you can rebase (not sure if it was before). merge on green.

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**